src/Settings.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace JsonI18n;
+
+/**
+ * Internationalization class to handle date formatting
+ * @author Vincent Diep
+ * @license https://www.mozilla.org/en-US/MPL/2.0/ MPL-2.0
+ */
+class Settings
+{
+    /**
+     * The type of fallback
+     * @var bool
+     */
+    protected $fallbackWithKey;
+
+    /**
+     * Creates a new JsonI18n\Settings instance
+     * @param array $settings The custom settings
+     * @throws \InvalidArgumentException If the locale parameter is empty
+     */
+    public function __construct(array $settings = [])
+    {
+        if (!$settings) {
+            $this->setFallbackWithKey();
+            return;
+        }
+
+        if (isset($settings['fallbackWithKey'])) {
+            $fallbackWithKey = $settings['fallbackWithKey'];
+            if (is_bool($fallbackWithKey))
+                $this->setFallbackWithKey($fallbackWithKey);
+            else
+                throw new \InvalidArgumentException('Invalid fallback setting.');
+        } else {
+            $this->setFallbackWithKey();
+        }
```

Comment:
This is a good approach but I think it can be a bit more lenient:

```suggestion
        $defaultSettings = ['strict' => true];
        $settings = array_merge($defaultSettings, $settings);
        
        $this->strict = (bool) $settings['strict'];
```

The suggestion also assumes my comment above on calling the option `strict`, but that can be discussed further.

Reply:
Great solution !